Makefile

```diff
@@ -565,3 +572,48 @@ octavia_deploy_cleanup: ## cleans up the service instance, Does not affect the o
 	oc kustomize ${DEPLOY_DIR} | oc delete --ignore-not-found=true -f -
 	rm -Rf ${OPERATOR_BASE_DIR}/octavia-operator ${DEPLOY_DIR}
 	oc rsh -t mariadb-openstack mysql -u root --password=${PASSWORD} -e "drop database octavia;" || true
+
+##@ NOVA
+.PHONY: nova_prep
+nova_prep: export IMAGE=${NOVA_IMG}
+nova_prep: ## creates the files to install the operator using olm
+	$(eval $(call vars,$@,nova))
+	bash scripts/gen-olm.sh
+
+.PHONY: nova
+nova: namespace nova_prep ## installs the operator, also runs the prep step. Set NOVA_IMG for custom image.
+	$(eval $(call vars,$@,nova))
+	oc apply -f ${OPERATOR_DIR}
+
+.PHONY: nova_cleanup
+nova_cleanup: ## deletes the operator, but does not cleanup the service resources
+	$(eval $(call vars,$@,nova))
+	bash scripts/operator-cleanup.sh
+	rm -Rf ${OPERATOR_DIR}
+
+.PHONY: nova_deploy_prep
+nova_deploy_prep: export KIND=Nova
+# TOOD(gibi): the tooling expect a containerImage at the top level
+# but for projects like Cinder and Nova where there are multiple services with
+# different images this customization does not make sense. Make this
+# customization optional in the tooling.
+nova_deploy_prep: export IMAGE=unused
+nova_deploy_prep: nova_deploy_cleanup ## prepares the CR to install the service based on the service sample file NOVA
+	$(eval $(call vars,$@,nova))
+	mkdir -p ${OPERATOR_BASE_DIR} ${OPERATOR_DIR} ${DEPLOY_DIR}
+	pushd ${OPERATOR_BASE_DIR} && git clone -b ${NOVA_BRANCH} ${NOVA_REPO} && popd
+	cp ${NOVA_CR} ${DEPLOY_DIR}
+	bash scripts/gen-service-kustomize.sh
+
+.PHONY: nova_deploy
+nova_deploy: input nova_deploy_prep ## installs the service instance using kustomize. Runs prep step in advance. Set NOVA_REPO and NOVA_BRANCH to deploy from a custom repo.
+	$(eval $(call vars,$@,nova))
+	oc kustomize ${DEPLOY_DIR} | oc apply -f -
```

Comment:
by the way i think 
oc kustomize ${DEPLOY_DIR} | oc apply -f -
could be 
oc apply -k ${DEPLOY_DIR} but i have not tried that.

kustomize is integrated into kubctl/oc in more recent version i belive so we likely can simplify this in the future if we want too.